qiskit/__init__.py

```diff
@@ -18,6 +18,15 @@
 import sys
 import warnings
 
+import qiskit._accelerate
+
+# Globally define compiled modules. The normal import mechanism will not
+# find compiled submodules in _accelerate because it relies on file paths
+# manually define them on import so people can directly import
+# qiskit._accelerate.* submodules and not have to rely on attribute access
+sys.modules["qiskit._accelerate.stochastic_swap"] = qiskit._accelerate.stochastic_swap
```

Comment:
It's a limitation of compiled extensions in general. Python's importer doesn't know how to search for submodules in a compiled extension. It's just extra annoying from rust because it doesn't give you a way to expose multiple binary libs from a single library as it expects people to use workspaces to do multiple libraries so we're stuck either doing this or creating a separate workspace (which means a separate dir with a standalone cargo.toml) which would have a lot of disadvantages.